pkg/kudoctl/util/resources/upgrade.go

```diff
@@ -0,0 +1,63 @@
+package resources
+
+import (
+	"fmt"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1alpha1"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/clog"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/kudo"
+	util "github.com/kudobuilder/kudo/pkg/util/kudo"
+
+	"github.com/Masterminds/semver"
+)
+
+// UpgradeOperatorVersion upgrades an OperatorVersion and its Instance.
+// For the updated Instance, new parameters can be provided.
+func UpgradeOperatorVersion(kc *kudo.Client, newOv *v1alpha1.OperatorVersion, instanceName, namespace string, parameters map[string]string) error {
```

Comment:
We already have `kudo.Client` with following methods:
```
func (c *Client) OperatorExistsInCluster()
func (c *Client) InstanceExistsInCluster()
func (c *Client) GetInstance() 
func (c *Client) GetOperatorVersion()
func (c *Client) UpdateInstance()
func (c *Client) ListInstances()
func (c *Client) OperatorVersionsInstalled()
func (c *Client) InstallOperatorObjToCluster()
func (c *Client) InstallOperatorVersionObjToCluster()
func (c *Client) InstallInstanceObjToCluster()
func (c *Client) DeleteInstance()
func (c *Client) ValidateServerForOperator()
```
which roughly encapsulate our business logic of handling KUDO's custom resources. Wouldn't this method be better placed there?

Reply:
Depends on what the scope of `kudo.Client` should be. Most functions there currently only call a function of `clientset` and add some error wrapping and/or format input/output types to better fit the ones we're using. The exception is `ValidateServerForOperator` which does a bit more. Both, `Install` as well as `UpgradeOperatorVersion` could be in the `kudo` package instead, if its scope isn't to be a wrapper for `clientset` calls.